aws/src/main/java/org/apache/iceberg/aws/LakeFormationAwsClientFactory.java

```diff
@@ -0,0 +1,199 @@
+      Preconditions.checkArgument(region == null || region.isEmpty(),
+          "Cannot initialize LakeFormationCredentialsProvider with empty region");
+      client = LakeFormationClient.builder()
+          .credentialsProvider(awsCredentialsProvider)
+          .region(Region.of(region)).build();
+      tableArn = getTableArn(properties);
+    }
+
+    @Override
+    public AwsCredentials resolveCredentials() {
+      GetTemporaryGlueTableCredentialsRequest getTemporaryGlueTableCredentialsRequest =
+          GetTemporaryGlueTableCredentialsRequest.builder()
+              .tableArn(tableArn)
+              .permissions(Permission.ALL)
+              .build();
+      GetTemporaryGlueTableCredentialsResponse response =
+          client.getTemporaryGlueTableCredentials(getTemporaryGlueTableCredentialsRequest);
+      return AwsSessionCredentials.create(response.accessKeyId(), response.secretAccessKey(), response.sessionToken());
+    }
+
+    private static AwsCredentialsProvider buildAssumeRoleCredentialsProvider(Map<String, String> properties) {
+      String roleArn = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_ARN);
+      Preconditions.checkNotNull(roleArn, "Cannot initialize LakeFormationCredentialsProvider with empty roleArn");
+      String sessionName = String.format("iceberg-aws-%s", UUID.randomUUID());
+      String externalId = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_EXTERNAL_ID);
+      Set<Tag> tag = toTags(properties);
+      Preconditions.checkArgument(tag.stream().anyMatch(t -> t.key().equals(LF_AUTHORIZED_CALLER_KEY)),
+          "must set LakeFormationAuthorizedCaller through client.assume-role.tags");
+      int timeout = PropertyUtil.propertyAsInt(properties,
+          AwsProperties.CLIENT_ASSUME_ROLE_TIMEOUT_SEC, AwsProperties.CLIENT_ASSUME_ROLE_TIMEOUT_SEC_DEFAULT);
+      AssumeRoleRequest request = AssumeRoleRequest.builder()
+          .roleArn(roleArn)
+          .roleSessionName(sessionName)
+          .durationSeconds(timeout)
+          .externalId(externalId)
+          .tags(tag)
+          .build();
+      return StsAssumeRoleCredentialsProvider.builder()
+          .stsClient(StsClient.builder().httpClientBuilder(UrlConnectionHttpClient.builder()).build())
+          .refreshRequest(request)
+          .build();
+    }
+
+    private static Set<Tag> toTags(Map<String, String> properties) {
+      return PropertyUtil.propertiesWithPrefix(properties, AwsProperties.CLIENT_ASSUME_ROLE_TAGS_PREFIX)
+          .entrySet().stream()
+          .map(e -> Tag.builder().key(e.getKey()).value(e.getValue()).build())
+          .collect(Collectors.toSet());
+    }
+
+    private String getTableArn(Map<String, String> properties) {
+      String dbName = properties.get(AwsProperties.LAKE_FORMATION_DB_NAME);
+      String tableName = properties.get(AwsProperties.LAKE_FORMATION_TABLE_NAME);
+      String accountId = properties.get(AwsProperties.LAKE_FORMATION_ACCOUNT_ID);
```

Comment:
I think we should use `glue.catalog-id` here and check that the value must not be null, instead of using a new account id property. 

Consider from the third-party engine's perspective, the catalog ID must be the user's account ID. Given LF is an extension of Glue, and here we are just constructing a Glue table ARN, I think it makes sense to use that instead of a different input.

Reply:
Discussed offline, we basically have 4 ways:
1. use Glue catalog ID (`glue.id`) and enforce catalog ID not null, which is a bit confusing although Glue as of today always use account ID as the catalog ID. This might change in the future.
2. derive account ID from assume role ARN. The role might be in a different account and does not work
3. use a new config `lakeformation.account-id`, which does not feel very correct because the account ID is used to set Glue table ARN
4. use a new config `glue.account-id`, we agreed this is the best way to both show it's a part of Glue information, and also distinguish from Glue catalog ID.